src/protocol/handshake.rs

```diff
@@ -0,0 +1,107 @@
+use std::io;
+
+use bytes::Bytes;
+use futures_util::{sink::SinkExt, stream::TryStreamExt};
+use pea2pea::{protocols::Handshake, Connection, ConnectionSide, Pea2Pea};
+use tokio_util::codec::{BytesCodec, Framed};
+use tracing::*;
+
+use crate::tools::inner_node::InnerNode;
+
+// Info from RFC 6455, section 4.1, page 18:
+//
+// The request MUST include a header field with the name
+// |Sec-WebSocket-Key|.  The value of this header field MUST be a
+// nonce consisting of a randomly selected 16-byte value that has
+// been base64-encoded (see Section 4 of [RFC4648]).  The nonce
+// MUST be selected randomly for each connection.
+//
+// NOTE: As an example, if the randomly selected value was the
+// sequence of bytes 0x01 0x02 0x03 0x04 0x05 0x06 0x07 0x08 0x09
+// 0x0a 0x0b 0x0c 0x0d 0x0e 0x0f 0x10, the value of the header
+// field would be "AQIDBAUGBwgJCgsMDQ4PEC=="
+/// A key-accept pair for a Sec-WebSocket-Key header.
+struct SecWebSocket {
+    key: String,
+    accept: String,
+}
+
+impl SecWebSocket {
+    /// Generate key-accept pair for a WebSocket handshake.
+    fn generate() -> Self {
+        // TODO(Rqnsom): Randomize 16 bytes.
+        let key: Vec<u8> = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
```

Comment:
we should probably use a more randomized key to have greater coverage in the tests

Reply:
ah, it's a TODO :+1: